frontend/src/components/CredentialTile.vue

```diff
@@ -0,0 +1,162 @@
+              <li v-for="{title} in phase.unpreparedRotations" :key="title">{{title}}</li>
+            </ul>
+          </template>
+          <template v-else>
+            <div>
+              This two-step operation is in phase
+              <v-chip color="primary" label x-small class="ml-2">Prepared</v-chip>
+            </div>
+            <div>
+              Rotation Prepared: <time-string :date-time="lastInitiationTime" mode="past"></time-string>
+            </div>
+          </template>
+        </v-tooltip>
+      </v-list-item-title>
+      <v-list-item-subtitle class="d-flex align-center">
+        <template v-if="type === 'certificateAuthorities' && !isCACertificateValiditiesAcceptable">
+          <shoot-messages :shoot-item="shootItem" :filter="['cacertificatevalidities-constraint']" small class="mr-1" />
+          <span color="warning">Certificate Authorities will expire in less than one year</span>
+        </template>
+        <template v-else>
+          <span v-if="!!lastCompletionTime">Last Rotated: <time-string :date-time="lastCompletionTime" mode="past"></time-string></span>
+          <span v-else>Not yet rotated</span>
+        </template>
+      </v-list-item-subtitle>
+    </v-list-item-content>
+    <v-list-item-action :class="{'py-0 my-1' : dense}">
+      <rotate-credentials :shoot-item="shootItem" :type="type"></rotate-credentials>
+    </v-list-item-action>
+  </v-list-item>
+</template>
+
+<script>
+import RotateCredentials from '@/components/RotateCredentials'
+import TimeString from '@/components/TimeString'
+import ShootMessages from '@/components/ShootMessages/ShootMessages'
+import { shootItem } from '@/mixins/shootItem'
+import get from 'lodash/get'
+import flatMap from 'lodash/flatMap'
+import head from 'lodash/head'
+import compact from 'lodash/compact'
+import { rotationTypes } from '@/utils/credentialsRotation'
+
+export default {
+  name: 'credential-tile',
+  components: {
+    RotateCredentials,
+    TimeString,
+    ShootMessages
+  },
+  props: {
+    type: {
+      type: String,
+      default: 'allCredentials',
+      required: false
+    },
+    dense: {
+      type: Boolean,
+      required: false
+    }
+  },
+  mixins: [shootItem],
+  computed: {
+    rotationStatus () {
+      return get(this.shootStatusCredentialsRotation, this.type, {})
+    },
+    lastInitiationTime () {
+      if (this.type !== 'allCredentials') {
+        return this.rotationStatus.lastInitiationTime
+      }
+      // Do not show aggregated initiation time
+      return undefined
+    },
+    lastCompletionTime () {
+      if (this.type !== 'allCredentials') {
+        return this.rotationStatus.lastCompletionTime
+      }
+      const allCompletionTimes = compact(flatMap(this.shootStatusCredentialsRotation, 'lastCompletionTime')).sort()
+      let requiredNumberOfRotationTimes = Object.keys(rotationTypes).length - 1 // There is no "all credentials" rotation type in the rotation status
```

Comment:
Either remove `allCredentials` from the `rotationTypes` list and handle this case separately. Maybe you can by this means get rid of the `if (this.type !== 'allCredentials') {` checks?

or tag this "synthetic" item so that you can filter it out

Reply:
Well it is no "synthetic item" its an actual rotation type. However it does not have it's own entry in the status (BTW if it has one this would make life for us way easier)
But I agree that it is bad as of now I will try to improve the code also regarding your comment about the object type check pattern. This is kind of a leftover from when I tried different ways to handle this and could not come up with something better at that time. Now, with a bit more distance I'll try to improve the code.